webapp/application.py

```diff
@@ -125,4 +131,178 @@ def application_page(token):
         application=application,
         job=job,
         hiring_lead=hiring_lead,
+        token=token,
+    )
+
+
+@application.route("/withdraw/<string:token>")
+def application_withdrawal(token):
+    payload = verification_token_cipher.decrypt(token)
```

Comment:
I would imagine that failing to decrypt this cipher must be fairly likely. Should we be adding a `try` block here so we can display a more friendly error page than just a plain 500 when a there's a problem with the token?